microsoft365defender_connector.py

```diff
@@ -1150,6 +1150,164 @@ def _handle_update_alert(self, param):
+            # Taking current UTC time as end time
+            end_time = datetime.utcnow()
+            ret_val = self._check_invalid_since_utc_time(action_result, time)
+            if phantom.is_fail(ret_val):
+                return action_result.set_status(phantom.APP_ERROR, LOG_UTC_SINCE_TIME_ERROR)
+            # Checking future date
+            if time >= end_time:
+                message = LOG_GREATER_EQUAL_TIME_ERROR.format(LOG_CONFIG_TIME_POLL_NOW)
+                return action_result.set_status(phantom.APP_ERROR, message)
+        except Exception as e:
+            message = "Invalid date string received. Error occurred while checking date format. Error: {}".format(str(e))
+            return action_result.set_status(phantom.APP_ERROR, message)
+        return phantom.APP_SUCCESS
+
+    def _handle_on_poll(self, param):
+        action_result = self.add_action_result(ActionResult(dict(param)))
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+        config = self.get_config()
+
+        # params for list incidents
+        poll_filter, offset, orderby = config.get(DEFENDER_INCIDENT_FILTER, ""), 0, "lastUpdateDateTime"
+        start_time_scheduled_poll = config.get(DEFENDER_CONFIG_START_TIME_SCHEDULED_POLL)
+        last_modified_time = (datetime.now() - timedelta(days=7)).strftime(DEFENDER_APP_DT_STR_FORMAT)  # Let's fall back to the last 7 days
+
+        if start_time_scheduled_poll:
+            ret_val = self._check_date_format(action_result, start_time_scheduled_poll)
+            # if date format is not valid
+            if phantom.is_fail(ret_val):
+                self.save_progress(action_result.get_message())
+                return action_result.set_status(phantom.APP_ERROR)
+
+            # set start time as the last modified time to, hence data is fetched from that point.
+            last_modified_time = start_time_scheduled_poll
+
+        if self.is_poll_now():
+            max_incidents = int(param.get(phantom.APP_JSON_CONTAINER_COUNT))
+        else:
+            max_incidents = config.get(DEFENDER_CONFIG_FIRST_RUN_MAX_INCIDENTS,
+                                       DEFENDER_INCIDENT_DEFAULT_LIMIT_FOR_SCHEDULE_POLLING)
+            ret_val, max_incidents = self._validate_integer(action_result, max_incidents, "max_incidents")
+            if phantom.is_fail(ret_val):
+                return action_result.get_status()
+
+            if self._state.get(STATE_FIRST_RUN, True):
+                self._state[STATE_FIRST_RUN] = False
+            elif last_time := self._state.get(STATE_LAST_TIME):
+                last_modified_time = last_time  # noqa: F841
+
+        start_time_filter = f"lastUpdateDateTime ge {last_modified_time}"
+        poll_filter += start_time_filter if not poll_filter else f" and {start_time_filter}"
+
+        endpoint = "{0}{1}".format(DEFENDER_MSGRAPH_API_BASE_URL, DEFENDER_LIST_INCIDENTS_ENDPOINT)
+        endpoint += "?$expand=alerts"
```

Comment:
```suggestion
        endpoint = "{0}{1}?$expand=alerts".format(DEFENDER_MSGRAPH_API_BASE_URL, DEFENDER_LIST_INCIDENTS_ENDPOINT)
```
Needs formatting though

Reply:
Done